### ISSUE TITLE
Volla 22 and X23 crash repeatedly after upgrade to VOS13

### ISSUE THREAD

**Issue by franco-bez** [HelloVolla/volla-os-beta-test#153]
There are several reports in the TG group that the securitymode configuration for blacklisting has bad sideeffects on a volla 22.
The phone crashes and restarts repeatedly with security mode turned on for blacklisting using templates.

On my X23 the same usage scenario works fine, no issues.

https://t.me/hello_volla/108035
https://t.me/hello_volla/107972
https://t.me/hello_volla/107959

**Comment by Affel24** [comment]
We have the same issue on two volla 22. The phone crashed sometimes while using, but also while the screen was just off. The problem definitely always appeared, when deleting all the app data for security mode and setting the blacklisted domains and activate it. Its not always crashing immediately, sometimes it takes a few moments.

After the security mode has been set and activated for first time, it continues to crash, even if the security mode is deactivated.
It seems the crashes are happening in irregular moments.

**Comment by wurzer** [comment]
I had observed a reboot once when I had accessed the security mode settings under Volla OS 13 on my Volla Phine 22 as a daily companion. However, I can no longer reproduce this behavior. 

I have now removed all templates again and downloaded them again. It could possibly be related to the automatic updates of the templates that we had introduced. Were you already using the templates under Volla OS 12?

How exactly do you go about observing a restart?
Have you made a configuration for the app firewall?

**Comment by wurzer** [comment]
I made a test today by forcing an update of the whitelist. I also modified the app firewall. Unfortunately I can't reproduce a crash. Looks like any other backbround process is causing this issue. 

Can you describe the steps to reproduce this issue?

**Comment by franco-bez** [comment]
I can not reproduce the problem myself as the bug does not affect my X23. I had Securitymode with blacklists (all templates activated) already active under VOS12.

The reports from the TG group seem to related to blacklisting, not whitelisting.

I can only ask on TG for affected user to report more details.

**Comment by franco-bez** [comment]
Some feedback in the TG group
https://t.me/hello_volla/108138

**Comment by wurzer** [comment]
I can't reproduce the issue. Does someone has steps to reproduce it? 

1. Removing all templates
2. Resetting security mode settings app
3. Re-importing all black list templates
4. Switching security mode on
5. Switching security mode off
6. Modifying blocked app list
7. Modifying app firewall

**Comment by franco-bez** [comment]
Some testing has been done.
an unexpected bug showed up in the configuration.
1. Import a Template to the Blacklist
2. Change TAB i.e. to App Firewall and then back to Firewall
3. in the menu choose "remove all" 
4. leave the "Security mode Settings"
5. Open "Security Mode Settings" choose TAB Firewall.
6. The Template is back on the blacklist -> should be empty
Only way to really clear the blacklist is to "remove all" and then switch to whitelisting.

----------
The reboots occur by random as long as there are templates in the blacklist.
Does not matter if the Security mode is turned on or off. 
Might take a few Minutes, the reboot happens when surfing the web, taking pictures, changing system settings.
It's completely random.

**Comment by franco-bez** [comment]
Playing around with the Security Mode Settings on my X23
Suddenly it rebooted too :-(
I could reproduce the bug 100% on the X23 just like described in the previous post for a V22.

1. First I chose "remove all" from the blacklist
2. Then I left the "Security mode Settings"
3. Opened "Security Mode Settings" choose TAB Firewall. - here the list of templates was empty - so I thought the X23 was not affected.
4. I added all available templates to the blacklist, just like before.
5. turned Security mode on again.

Suddenly the phone rebooted.
Then it showed exactly the same behaviour as described in the previous post.

I had a hard time switching off the security mode and removing all templates as the phone rebooted 4 times until I was able to finish.

Now with an empty blacklist the phone seems to be stable again.

**Comment by franco-bez** [comment]
additional info:
the reboots are different from "normal" reboots, maybe only the GUI restarts.
The Volla Boot Manager is NOT shown.
The PIN numbers for the SIM cards are NOT needed.
The whole process is much faster than a "normal" reboot.

**Comment by wurzer** [comment]
Looks like it is a race condition of the app start and the app logic to check for updates of templates. We'll soon publish a beta version with a fix.

**Comment by Affel24** [comment]
I want to add this two screen videos to this issue.
One shows how the templates for the Blacklisted domains can not be deleted and the other one shows, that the templates are shown even twice in the list, instead of to be overwritten. Maybe this bugs are related to the crash?

Anyway, the only way to solve this at least temporary, is to delete all the app data and reconfigure the safety mode.
If I do so, then the System usually crashes. 

https://github.com/HelloVolla/volla-os-beta-test/assets/165715256/04cae4f6-7aaa-4419-b3d7-8a532cf6f0cf
https://github.com/HelloVolla/volla-os-beta-test/assets/165715256/bc70b1c1-082e-46a1-ae14-a6ca62b2ddb7

**Comment by franco-bez** [comment]
@Affel24 as I wrote in one of my initial posts, first clear the blacklist, then switch to "whitelisting" and then leave the configuration. This way the blacklist is really emptied.

Wie oben schon geschrieben wenn du die Templates entfernst zuerst auf "Einschlussliste Domänen" umschalten und dann erst zurück auf Einstellungen. Wenn man dann wieder reingeht und auf "Ausschlussliste Domänen" umschaltet, dann ist und bleibt sie leer.

**Comment by wurzer** [comment]
Will be fixed in the next update

**Comment by franco-bez** [comment]
13-20240430-STABLE-vidofnir I cannot confirm the fix
Situation unchanged when manipulating black list
https://github.com/HelloVolla/volla-os-beta-test/issues/153#issuecomment-2050542147

No reboot so far on my X23. 

New problem: the blacklisting seems to produce a sort of unstable filtering. 
"good" websites get blocked, sometimes you have to retry several times to get the page displayed in fennec. So it feels random whether a websites gets blocked or not.

I have all templates active in blacklisting.

**Comment by franco-bez** [comment]
13-20240503-STABLE-vidofnir still the same - the problem is not fixed
try playing around with the Blacklist settings after a few minutes doing so it will crash almost certainly

**Comment by franco-bez** [comment]
I made some more tests.
It seems to me that the blacklisting mode with all templates activated needs quite some time to "run in".
At first browsing the web was painful. It took up to 15 times taping on "try again" in order to get a wikipedia page displayed.
One of my devices even rebooted, but only once.
After several hours, browsing the web became smoother. 
Now it almost feels normal again.

I observed that the memory usage of the "security mode settings" app grew from initially 5 MB to 30 MB.

I also once tried to download all templates lists to my PC. Its a total of almost 43 MB plain text, more than 2 million lines of blacklisted domains.
I wanted to avoid the template update mechanism for a test, so I tried to import one of those files via "import list". This failed miserably. I do not know how many lines the "import lists" feature supports.


So I think the issue is not completely fixed.
- playing around with the Blacklist settings can still crash the OS.
- removing templates, or "clear all" does not reliably work.
- occasional restarts still happen, but much less frequently
